torchrec/distributed/train_pipeline.py

```diff
@@ -68,7 +69,7 @@ def _wait_for_batch(batch: In, stream: Optional[torch.cuda.streams.Stream]) -> N
     batch.record_stream(cur_stream)
 
 
-class TrainPipelineBase(TrainPipeline[In, Out]):
+class TrainPipelineBase(TrainPipeline[In, Out], Iterable[Out]):
     """
     This class runs training iterations using a pipeline of two stages, each as a CUDA
     stream, namely, the current (default) stream and `self._memcpy_stream`. For each
```

Comment:
please add args to the docstrings.

let's also clarify in the docstring (via example) that this can be used as iterator API via next (with passed in data-laoder_iter) , or via progress().

Reply:
There is no really good docs right now... We are actually discussing a new fully componentized pipeline API, which will allow me to fully keep my original training loop. Let's document it very well as a part of this new API.

```
torchrec.dist.pipeline_data(data_pipe: IterDataPipe, model: nn.Module) -> Tuple[IterDataPipe, nn.Module]


data_pipe, model = torchrec.dist.pipeline_data(input_loader, model)

for batch in data_pipe:
  opt.zero_grad()
  res = model(batch)
  res.backward()
  opt.step()
```